src/app/controllers/workday/create.js

```diff
@@ -0,0 +1,179 @@
+            }
+        });
+}
+
+var loadWorkTypes = function($scope, $log, messenger, oauthRequest) {
+    oauthRequest.get('http://localhost:50323/api/v1/worktypes/simple')
+        .then(function success(response) {
+
+            $scope.data.worktypes = response.data;
+
+        }, function error(response) {
+
+            $log.error(response);
+
+            if (response.status === 401) {
+                messenger.unauthorized($scope);
+            } else {
+                messenger.showErrorMessage($scope, { error: response });
+            }
+
+        });
+}
+
+var configureSubmit = function($scope, $log, messenger, oauthRequest) {
+
+    $scope.submit = function() {
+        $scope.loading = true;
+
+        var msg = {
+            locationId: $scope.workday.location,
+            travelTimeHours: $scope.workday.travelHours,
+            workDate: $scope.workday.workDate,
+            workHours: $scope.workday.workHours,
+            workTypeId: $scope.workday.workType,
+            userId: 1 // need to extract from token
+        };
+
+        oauthRequest.post('http://localhost:50323/api/v1/worklogs', msg)
+           .then(function success(response){
+                messenger.showSuccessMessage($scope, 'Saved');
+                $scope.loading = false;
+           }, function error(response) {
+                switch (response.status) {
+                    case 400:
+                        var messages = [{ error: response.data.message }];
+                        if (response.data.modelState) {
+                            _.forIn(response.data.modelState, function (value, key) {
+                                messages.push({ error: value.toString() });
+                            });
+                        }
+                        messenger.showErrorMessage($scope, messages);
+                        break;
+                    case 401:
+                        messenger.unauthorized($scope);
+                        break;
+                    default:
+                        messenger.showErrorMessage($scope, { error: response });
+                }
+                $scope.loading = false;
+           });
+    }
+
+    return $scope;
+}
+
+/**
+ * @ngdoc function
+ * @name flightNodeApp.controller:WorkdayCreateController
+ * @description
+ * # WorkdayController
+ * Controller for the workday logging page.
+ */
+angular.module('flightNodeApp')
+    .controller('WorkdayCreateController',
+        ['$scope', '$location', '$http', '$log', 'messenger', 'oauthRequest',
+            function ($scope, $location, $http, $log, messenger, oauthRequest) {
+                $scope.loading = true;
+                $scope.data = {};
+
+                $scope = configureDateField($scope);
+                loadLocations($scope, $log, messenger, oauthRequest);
+                loadWorkTypes($scope, $log, messenger, oauthRequest);
+
+                $scope = configureSubmit($scope, $log, messenger, oauthRequest);
```

Comment:
This would probably be cleaner as 

    $scope.submit = configureSubmit($scope, $log, messenger, oauthRequest);